rules/prefer-string-raw.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+const {isStringLiteral, isDirective} = require('./ast/index.js');
+const {fixSpaceAroundKeyword} = require('./fix/index.js');
+
+const MESSAGE_ID = 'prefer-string-raw';
+const messages = {
+	[MESSAGE_ID]: '`String.raw` should be used to avoid escaping `\\`.',
+};
+
+const BACKSLASH = '\\';
```

Comment:
Shouldn't these use `String.raw`?

Reply:
``String.raw`\` `` is an error and ``String.raw`\\` `` products two backslash, so it's impossible to create a a single backslash with `String.raw`.

We can't do ``String.raw`foo\` `` either. That's why we ignore string ends with backslash  https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2339/files#diff-def49face997bfe9729f748b6720c2847611a4ad3b6a9714a96f0ef408259d75R65

Reply:
The one in `messages` can't use `String.raw` either, because it contains `` ` ``, `` String.raw`foo`bar` `` this is invalid code. Ignored in code too https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2339/files#diff-def49face997bfe9729f748b6720c2847611a4ad3b6a9714a96f0ef408259d75R57

Reply:
Ah ok.